scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+            evaluated.
+        method : {None, 'formula', 'logexp'}
+            The strategy used to evaluate the log-PDF. By default (``None``), the
+            infrastructure chooses between the following options.
+
+            - ``'formula'``: use a formula for the log-PDF itself
+            - ``'logexp'``: evaluate the PDF and takes its logarithm
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The log-PDF evaluated at the argument `x`.
+
+        See Also
+        --------
+        ContinuousDistribution.pdf
+        ContinuousDistribution.logcdf
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`.
+        By definition of the support, the log-PDF evaluates to its minimum value
+        of :math:`-\infty` (i.e. :math:`\log(0)`) outside the support; i.e. for
+        :math:`x < l` or :math:`x > r`. The maximum of the log-PDF may be less
+        than or greater than :math:`\log(1) = 0` because the maximum of the PDF
+        can be any positive real.
+
+        For distributions with infinite support, it is common for
+        `ContinuousDistribution.pdf` to return a value of `0` when the argument
+        is theoretically within the support; this can occur because the true value
+        of the PDF is too small to be represented by the chosen dtype. The log
+        of the PDF, however, will often be numerically nonzero over a much larger
+        domain. Consequently, it may be preferred to work with the logarithms of
+        probabilities and probability densities to avoid underflow.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-1.0, b=1.0)
+
+        Evaluate the log-PDF at the desired argument:
+
+        >>> X.logpdf(0.5)
+        -0.6931471805599453
+        >>> np.allclose(X.logpdf(0.5), np.log(X.pdf(0.5)))
+        True
+
+        """
+        return self._logpdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _logpdf_dispatch(self, x, *, method=None, **kwargs):
+        if self._overrides('_logpdf_formula'):
+            method = self._logpdf_formula
+        elif self.tol is _null:  # ensure that developers override _logpdf
+            method = self._logpdf_logexp
+        return method
+
+    def _logpdf_formula(self, x, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _logpdf_logexp(self, x, **kwargs):
+        return np.log(self._pdf_dispatch(x, **kwargs))
+
+    @_set_invalid_nan
+    def pdf(self, x, *, method=None):
+        r"""Probability density function
+
+        The probability density function :math:`f(x)` is the probability *per
```

Comment:
not sure what is meant by unit interval

I am wondering if it is easier to say that `P[X in (a, b)] = \int_a^b f(u) du`

Reply:
> `P[X in (a, b)] = \int_a^b f(u) du`

Is not a drop-in replacement for the sentence in question:

> The probability density function :math:`f(x)` is the probability *per unit interval* of :math:`x` that the random variable takes on the value :math:`x`.

The existing statement use "per unit ___" in the same way as the definition of [mass density](https://en.wikipedia.org/wiki/Density) defines it as "a substance's mass per unit of volume". Instead of replacing the statement entirely, consider suggesting a replacement for "per unit interval of :math:`x`".

Reply:
TBH, the unit interval wording did confuse me on a first read, too. How about @chrisb83 's version, possibly without an integral, for an infinitesimal interval dP = f(x) dx or some such.

Reply:
I am attempting to provide an English definition before a mathematical definition. Are you asking for me to:

1. replace the English definition with a second mathematical definition, essentially making this a statement of the fundamental theorem of calculus, or 
2. to replace the the definition of the PDF as the derivative of the CDF with an implicit definition of the PDF as that whose integral is the CDF?

Reply:
Probability per unit interval of $x$ sounds confusing to me too. I don't think there's a way to describe it without at least an intuitive understanding of concepts from calculus. You don't necessarily have to use the language of calculus, but I don't have a good suggestion yet. If we're able to find a good "plain English" sentence, I'd suggest making the mathematical definition a more precise version of that particular statement.

Reply:
Just changed it to "per unit length" per discussion with @steppi.

Reply:
It think it is difficult to explain it intuitively. Looking at the Wikipedia page (https://en.wikipedia.org/wiki/Probability_density_function), the best description i found there is "The PDF is used to specify the probability of the random variable falling within a particular range of values"

Reply:
Maybe it's my engineering background, but I'm having a really hard time seeing the difficulty here. If one understands the relationship between mass density and mass, I think the relationship between probability density and probability is immediately intuitive and easy to describe. To help me understand better where the difficulty is, could you answer the following questions?

- Is the relationship between mass density and mass intuitive?
- Is mass density in 1d not adequately described as "mass per unit length"?
- Is my assertion that the relationships are identical incorrect?

We can discuss this at the meeting.

Reply:
@tupui tell me what to do